.github/workflows/coverage.yml

```
@@ -0,0 +1,20 @@
+name: Test and coverage
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.21'
+      - name: Run coverage
+        run: go test -race -coverprofile=coverage.out -covermode=atomic ./...
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
```

```suggestion
        uses: codecov/codecov-action@v4
```

You can update some actions
https://github.com/codecov/codecov-action

Moreover, you can add dependabot
https://gist.github.com/magnetikonline/6f215db058e327905bce66c37f92426c
https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot